libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/util/CacheResultUtils.kt

```diff
@@ -17,6 +19,38 @@ internal object CacheResultUtils {
         override fun invoke(f: (P1, P2) -> R) = f(p1, p2)
     }
 
+    /**
+     * Specialized cache key that can be used for managing results of processing related to a route.
+     *
+     * This key uses [NavigationRoute.id] instead of [NavigationRoute.hashCode] because the latter produces a hash of the full [DirectionsRoute]
+     * which for longer routes and use cases that require very frequent access (multiple times per second) proves to be inefficient (refs https://github.com/mapbox/navigation-sdks/issues/1918).
```

Comment:
It's an internal issue, is it OK to place the link here?

Reply:
That's okay, it's an internal class and the repo itself has a generic name.